standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/parser/ExpressionTree.java

```diff
@@ -456,19 +457,22 @@ private String getJdoFilterPushdownParam(int partColIndex,
       boolean isIntegralSupported = canPushDownIntegral && canJdoUseStringsWithIntegral();
       String colType = partitionKeys.get(partColIndex).getType();
       // Can only support partitions whose types are string, or maybe integers
-      // Date data type value is considered as string hence pushing down to JDO.
+      // Date/Timestamp data type value is considered as string hence pushing down to JDO.
       if (!colType.equals(ColumnType.STRING_TYPE_NAME) && !colType.equals(ColumnType.DATE_TYPE_NAME)
+          && !colType.equals(ColumnType.TIMESTAMP_TYPE_NAME)
```

Comment:
Can `colType` be `null` here? Does case sensibility matters?
Please consider swapping the operands like
```
ColumnType.TIMESTAMP_TYPE_NAME.equalsIgnoreCase(colType)
```

Reply:
I don't think colType can be null ever.

Ack.